### PR TITLE
Prn

### PR DESCRIPTION
**Pull request by ganusyak**

### CODE REVIEW

File: src/main/java/com/cvtalks/PRNCalculator.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        // Returns an instance of binary operation
+        Operation operation = null;
+        switch (operationString) {
+            case "sin": operation = new Sin(value);
+                break;
+            case "cos": operation = new Cos(value);
+                break;
+            case "sqrt": operation = new Sqrt(value);
+                break;
+            default: throw new UnsupportedOperationException("Operation did not implemented yet");
```
Comment (BloodAxe) [inline src/main/java/com/cvtalks/PRNCalculator.java]: Please include operation name (operationString) into exception message.

File: src/main/java/com/cvtalks/PRNCalculator.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+                break;
+            case "-": operation = new Subtract(value1, value2);
+                break;
+            case "*": operation = new Multiply(value1, value2);
+                break;
+            case "/": operation = new Divide(value1, value2);
+                break;
+            case "^": operation = new Pow(value1, value2);
+                break;
+            default: throw new UnsupportedOperationException("Operation did not implemented yet");
```
Comment (BloodAxe) [inline src/main/java/com/cvtalks/PRNCalculator.java]: Please include operation name (operationString) into exception message.

File: src/main/java/com/cvtalks/PRNCalculator.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    }
+
+    private Operation constant(String operationString) {
+        Operation operation = null;
+        switch (operationString) {
+            case "pi": operation = new Value(Math.PI);
+                break;
+            case "e": operation = new Value(Math.E);
+                break;
+            default: throw new UnsupportedOperationException("Operation did not implemented yet");
```
Comment (BloodAxe) [inline src/main/java/com/cvtalks/PRNCalculator.java]: Please include operation name (operationString) into exception message.

File: src/main/java/com/cvtalks/PRNCalculator.java (hunk 1 above)
Comment (ganusyak) [inline src/main/java/com/cvtalks/PRNCalculator.java]: Done

File: src/main/java/com/cvtalks/PRNCalculator.java (hunk 2 above)
Comment (ganusyak) [inline src/main/java/com/cvtalks/PRNCalculator.java]: Done

File: src/main/java/com/cvtalks/PRNCalculator.java (hunk 3 above)
Comment (ganusyak) [inline src/main/java/com/cvtalks/PRNCalculator.java]: Done

File: src/main/java/com/cvtalks/OperatorParser.java [hunk 4]
```diff
@@ -19,4 +19,91 @@ public static boolean CheckBracesBalance (String input) {
...
+
+    public static boolean isParenthesis(String operand) {
+        return (operand.equals("(") || operand.equals(")"));
+    }
+
+    public static boolean isOperator(char operand) {
+        return ((operand == '+') || (operand == '-') || (operand == '*') || (operand == '/') || (operand == '^'));
+    }
+
+    public static String[] operationStringsFromString(String inputString) {
```
Comment (BloodAxe) [inline src/main/java/com/cvtalks/OperatorParser.java]: This method can be replaced by a single call to String.split() method: http://stackoverflow.com/questions/3481828/how-to-split-a-string-in-java

Comment (ganusyak) [inline src/main/java/com/cvtalks/OperatorParser.java]: implemented regex-based split

**Pull request merged**